htmd/qm/psi4.py

```diff
@@ -0,0 +1,244 @@
+        Point coordinates to compute ESP values. The array shape has to be (number_of_points, 3)
+    optimize : bool, default = False
+        Run geometery optimization
+    restrained_dihedrals : nunpy.ndarray, default = None
+        List of restrained dihedrals (0-based atom indices). The array shape has to be (number_of_restrained_dihedrals, 4).
+    queue : :class:`SimQueue`, defualt = :class:`LocalCPUQueue`
+        Queue object used to run simulations
+    directory : str, default = '.'
+        Working directoy
+
+    Examples
+    --------
+
+    Create an object of H2 molecule.
+    >>> import os
+    >>> from htmd.home import home
+    >>> from htmd.parameterization.ffmolecule import FFMolecule, FFTypeMethod
+    >>> molFile = os.path.join(home('test-qm'), 'H2-0.74.mol2')
+    >>> mol = FFMolecule(molFile, method=FFTypeMethod.NONE) # doctest: +ELLIPSIS
+    <BLANKLINE>
+    ...
+
+    Create a Psi4 object
+    >>> from htmd.qm import Psi4
+    >>> qm = Psi4()
+    >>> qm # doctest: +ELLIPSIS
+    <htmd.qm.psi4.Psi4 object at 0x...>
+
+    Run single-point QM calculation of H2 with BLYP and cc-pVDZ
+    >>> from tempfile import TemporaryDirectory
+    >>> with TemporaryDirectory() as tmp:
+    ...     qm.molecule = mol
+    ...     qm.theory = 'BLYP'
+    ...     qm.basis = 'cc-pVDZ'
+    ...     qm.directory = tmp
+    ...     result = qm.run()
+
+    The QM results are returned as a list of htmd.qm.QMResult objects. See htmd.qm.QMResult documentation for details.
+    >>> result # doctest: +ELLIPSIS
+    [<htmd.qm.base.QMResult object at 0x...>]
+    >>> result[0].errored
+    False
+    >>> result[0].energy # doctest: +ELLIPSIS
+    -728.97082622...
+    >>> result[0].mulliken
+    [0.0, -0.0]
+
+    Run the geometry optimization of H2 with BLYP, but change basis to 3-21G.
+    NOTE: the `directory` attribut needs to be set to empty or non-existing directory, overwise the previous
+    calculation results are read.
+    >>> with TemporaryDirectory() as tmp:
+    ...     qm.basis = '3-21G'
+    ...     qm.optimize = True
+    ...     qm.directory = tmp
+    ...     result = qm.run()
+
+    The initial coordinates and optimize coordinates of H2 can be compared
+    >>> mol.coords
+    array([[[ 0.  ],
+            [ 0.  ],
+            [-0.37]],
+    <BLANKLINE>
+           [[ 0.  ],
+            [ 0.  ],
+            [ 0.37]]], dtype=float32)
+    >>> result[0].coords
+    array([[[ 0.        ],
+            [ 0.        ],
+            [-0.37527978]],
+    <BLANKLINE>
+           [[ 0.        ],
+            [ 0.        ],
+            [ 0.37527978]]])
+
+    The QM calculations run using LocalCPUQueue by default, but this can be changed to the others.
+    >>> from htmd.queues.acecloudqueue import AceCloudQueue
+    >>> qm.queue = AceCloudQueue() # doctest: +SKIP
+    """
+
+    @property
+    def _command(self):
+        return 'export HTMD_PSI4_WORKDIR=$(pwd)\npsi4 -i psi4.in -o psi4.out &> psi4.log'
```

Comment:
What's this `HTMD_PSI4_WORKDIR`?

Reply:
Psi4 changes directory as mentioned later.